src/Observable.ts

```diff
@@ -96,9 +97,10 @@ export class Observable<T> implements CoreOperators<T>  {
       }
     } else {
       const next = <((x?) => void)> observerOrNext;
-      subscriber = Subscriber.create(next, error, complete);
+      subscriber = Subscriber.create(next, error, complete, start);
     }
 
+    subscriber.start(subscriber);
     subscriber.add(this._subscribe(subscriber));
```

Comment:
@blesh ah, I wasn't aware of that requirement. I'm thinking if subscribing is referentially transparent, and subscribing performs some action, then unsubscribing performs the inverse, then synchronous subscribe + unsubscribe is equivalent to not even subscribing in the first place. Does that sound right? What could the subscription do with a subscriber that's already disposed by the time it's called?